Jenkinsfile

```diff
@@ -72,22 +76,49 @@ pipeline {
                         }
                     }
                 }
-
-                stage('E2E Tests') {
-                    steps {
-                        sh '''
-                            docker-compose -f docker/docker-compose.yml -f docker/docker-compose-e2e-tests.yml build
-                            docker-compose -f docker/docker-compose.yml -f docker/docker-compose-e2e-tests.yml up --exit-code-from gp2gp-e2e-tests
-                            docker cp e2e-tests:/home/gradle/e2e-tests/build .
-                            mv build e2e-build
-                        '''
-                        archiveArtifacts artifacts: 'e2e-build/reports/**/*.*', fingerprint: true
-                        junit '**/e2e-build/test-results/**/*.xml'
+                stage('Deploy & Test') {
+                    options {
+                        lock("${tfProject}-${tfEnvironment}-${tfComponent}")
                     }
-                    post {
-                        always {
-                            sh "rm -rf e2e-build"
-                            sh "docker-compose -f docker/docker-compose.yml -f docker/docker-compose-e2e-tests.yml down"
+                    stages {
+
+                        stage('Deploy using Terraform') {
+                            steps {
+                                script {
+                                    String tfCodeBranch  = "develop"
+                                    String tfCodeRepo    = "https://github.com/nhsconnect/integration-adaptors"
+                                    String tfRegion      = "${TF_STATE_BUCKET_REGION}"
+                                    List<String> tfParams = []
+                                    Map<String,String> tfVariables = ["${tfComponent}_build_id": BUILD_TAG]
+                                    dir ("integration-adaptors") {
+                                      git (branch: tfCodeBranch, url: tfCodeRepo)
+                                      dir ("terraform/aws") {
+                                        if (terraformInit(TF_STATE_BUCKET, tfProject, tfEnvironment, tfComponent, tfRegion) !=0) { error("Terraform init failed")}
+                                        if (terraform('plan', TF_STATE_BUCKET, tfProject, tfEnvironment, tfComponent, tfRegion, tfVariables) !=0 ) { error("Terraform Apply failed")}
```

Comment:
change the error message to "Terraform Plan failed"

Reply:
Actually forgot to remove this line, this is updated now.